lib/ex_unit/lib/ex_unit/runner.ex

```diff
@@ -399,6 +399,14 @@ defmodule ExUnit.Runner do
     end)
   end
 
+  defp test_tags_context(test_tags, parent_context) do
+    Map.merge(parent_context, test_tags, fn
+      :tmp_dir, _v0, false -> false
+      :tmp_dir, v0, _v1 -> v0
+      _k, _v0, v1 -> v1
+    end)
+  end
```

Comment:
Can you elaborate on why we're doing this?

Reply:
So before the `context` was just `%{module: module, case: module}` and `Map.merge(test.tags, context)` was enough. Now `context` can contain `@moduletag` entries as well.

So initially I thought I should just swap the merge `Map.merge(context, test.tags)`, but that broke the tests that use `tmp_dir`. Because `maybe_create_tmp_dir` puts tmp_dir stuff inside the parent context it had to tag precedence over the test tags (e.g., `@tag :tmp_dir`), *but* not when it was set to false (you can have `@moduletag :tmp_dir` and then `@tag tmp_dir: false`.

If this is a but funky we could maybe clean this up / refactor by unifying the logic of creating the tmp dir with merging the tags.